plugin/plugin.go

```diff
@@ -1,243 +1,164 @@
-	Payload         []byte            `json:"payload"`
-	PluginConfig    map[string]string `json:"pluginConfig,omitempty"`
+func (p *CLIPlugin) GetMetadata(ctx context.Context, req *proto.GetMetadataRequest) (*proto.GetMetadataResponse, error) {
+	var resp proto.GetMetadataResponse
+	err := run(ctx, p.name, p.path, executor, req, &resp)
+	return &resp, err
 }
 
-func (GenerateSignatureRequest) Command() Command {
-	return CommandGenerateSignature
+func (p *CLIPlugin) DescribeKey(ctx context.Context, req *proto.DescribeKeyRequest) (*proto.DescribeKeyResponse, error) {
+	var resp proto.DescribeKeyResponse
+	err := run(ctx, p.name, p.path, executor, req, &resp)
+	return &resp, err
 }
 
-// GenerateSignatureResponse is the response of a generate-signature request.
-type GenerateSignatureResponse struct {
-	KeyID            string `json:"keyId"`
-	Signature        []byte `json:"signature"`
-	SigningAlgorithm string `json:"signingAlgorithm"`
-
-	// Ordered list of certificates starting with leaf certificate
-	// and ending with root certificate.
-	CertificateChain [][]byte `json:"certificateChain"`
+func (p *CLIPlugin) GenerateSignature(ctx context.Context, req *proto.GenerateSignatureRequest) (*proto.GenerateSignatureResponse, error) {
+	var resp proto.GenerateSignatureResponse
+	err := run(ctx, p.name, p.path, executor, req, &resp)
+	return &resp, err
 }
 
-// GenerateEnvelopeRequest contains the parameters passed in a generate-envelope request.
-type GenerateEnvelopeRequest struct {
-	ContractVersion       string            `json:"contractVersion"`
-	KeyID                 string            `json:"keyId"`
-	PayloadType           string            `json:"payloadType"`
-	SignatureEnvelopeType string            `json:"signatureEnvelopeType"`
-	Payload               []byte            `json:"payload"`
-	PluginConfig          map[string]string `json:"pluginConfig,omitempty"`
+func (p *CLIPlugin) GenerateEnvelope(ctx context.Context, req *proto.GenerateEnvelopeRequest) (*proto.GenerateEnvelopeResponse, error) {
+	var resp proto.GenerateEnvelopeResponse
+	err := run(ctx, p.name, p.path, executor, req, &resp)
+	return &resp, err
 }
 
-func (GenerateEnvelopeRequest) Command() Command {
-	return CommandGenerateEnvelope
+func (p *CLIPlugin) VerifySignature(ctx context.Context, req *proto.VerifySignatureRequest) (*proto.VerifySignatureResponse, error) {
+	var resp proto.VerifySignatureResponse
+	err := run(ctx, p.name, p.path, executor, req, &resp)
+	return &resp, err
 }
 
-// GenerateEnvelopeResponse is the response of a generate-envelope request.
-type GenerateEnvelopeResponse struct {
-	SignatureEnvelope     []byte            `json:"signatureEnvelope"`
-	SignatureEnvelopeType string            `json:"signatureEnvelopeType"`
-	Annotations           map[string]string `json:"annotations,omitempty"`
-}
+// NewCLIPlugin validate the metadata of the plugin and return a *CLIPlugin.
+func NewCLIPlugin(name, path string) (*CLIPlugin, error) {
```

Comment:
Should `NewCLIPlugin` take a `context.Context` as a parameter?

Reply:
Updated.